src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -1010,6 +1011,17 @@ static <R> R extractUniqueDrsObjectValue(
     }
   }
 
+  /** Given a list of DRSObjects, extract the first value and fail if there are no values */
+  @VisibleForTesting
+  static <R> R extractFirstDrsObjectValue(
+      List<DRSObject> drsObjects, Function<DRSObject, ? extends R> mapper) {
+    try {
+      return drsObjects.stream().map(mapper).findFirst().orElseThrow();
```

Comment:
It's probably fine to return the first, but you could also return all of them.
```suggestion
      return drsObjects.stream().map(mapper).collect(Collectors.joining(", "));
```

Reply:
Need to play with that some more, the object being mapped isn't necessarily a string